fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/pdagg/PushDownAggregateRewriter.java

```diff
@@ -0,0 +1,440 @@
+                        .forEach(v -> context.groupBys.put(v, v));
+
+                for (int i = 1; i < ifFn.getChildren().size(); i++) {
+                    if (ifFn.getChild(i).isConstant()) {
+                        Preconditions.checkState(ifFn.getChild(i).isConstantNull());
+                        ifFn.setChild(i, ConstantOperator.createNull(key.getType()));
+                        continue;
+                    }
+                    ColumnRefOperator ref = replaceByNewAggregation(aggFn, ifFn.getChild(i), context);
+                    ifFn.setChild(i, ref);
+                }
+
+                context.aggregations.remove(key);
+                originProjectMap.put(key,
+                        new CallOperator(ifFn.getFnName(), key.getType(), ifFn.getChildren(), ifFn.getFunction()));
+            } else {
+                context.aggregations.put(key, (CallOperator) rewriter.rewrite(aggFn));
+            }
+        }
+    }
+
+    private ColumnRefOperator replaceByNewAggregation(CallOperator originAggFn, ScalarOperator input,
+                                                      AggregateContext context) {
+        CallOperator newAgg = genAggregation(originAggFn, input);
+        ColumnRefOperator ref;
+        if (context.aggregations.containsValue(newAgg)) {
+            ref = context.aggregations.entrySet().stream().filter(e -> e.getValue().equals(newAgg))
+                    .findFirst().map(Map.Entry::getKey).orElseThrow(IllegalArgumentException::new);
+        } else {
+            ref = factory.create(newAgg, newAgg.getType(), newAgg.isNullable());
+        }
+        context.aggregations.put(ref, newAgg);
+        return ref;
+    }
+
+    @Override
+    public OptExpression visitLogicalAggregate(OptExpression optExpression, AggregateContext context) {
+        LogicalAggregationOperator aggregate = (LogicalAggregationOperator) optExpression.getOp();
+        if (!allRewriteContext.containsKey(aggregate)) {
+            return visit(optExpression, context);
+        }
+
+        List<AggregateContext> allRewrite = allRewriteContext.get(aggregate);
+        // rewrite
+        AggregateContext childContext = new AggregateContext();
+        childContext.origAggregator = aggregate;
+
+        Map<ColumnRefOperator, CallOperator> newAggregations = Maps.newHashMap(aggregate.getAggregations());
+
+        // flat aggregate
+        List<ColumnRefOperator> allAggregateRefs = allRewrite.stream()
+                .map(a -> a.aggregations.keySet())
+                .flatMap(Collection::stream)
+                .distinct().collect(Collectors.toList());
+
+        // rewrite origin aggregation
+        for (ColumnRefOperator ref : allAggregateRefs) {
+            CallOperator call = aggregate.getAggregations().get(ref);
+            ColumnRefOperator newRef = factory.create(call.getFnName(), call.getType(), call.isNullable());
+            childContext.aggregations.put(newRef, call);
```

Comment:
while childContext point to the original call rather than the newCall?

Reply:
like:
select max(v1) from t1;
=>
select max(x) from (select max(v1) as x from t1);